pkgs/by-name/ne/nexusmods-app/package.nix

```
@@ -44,23 +39,23 @@ buildDotnetModule rec {
 
   preConfigure = ''
     substituteInPlace Directory.Build.props \
-      --replace '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
+      --replace-fail '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
   '';
 
-  postPatch = ''
-    ln --force --symbolic "${lib.getExe _7zzWithOptionalUnfreeRarSupport}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
-  '';
+  NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR = 1; # Use nixpkgs 7zz executable
 
   makeWrapperArgs = [
     "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}"
     "--set APPIMAGE $out/bin/${meta.mainProgram}" # Make associating with nxm links work on Linux
+    "--set INSTALLATION_METHOD_PACKAGE_MANAGER 1" # Tell the app it's a distro package
   ];
 
   runtimeDeps = [
     fontconfig
     libICE
     libSM
     libX11
+    (lib.getExe (_7zz.override { inherit enableUnfree; }))
```

Correct me if I'm wrong, but I think you just pass the package?

```suggestion
    (_7zz.override { inherit enableUnfree; })
```